api/tests/routes/public/collective/endpoints/simulate_adage_steps/test_bookings.py

```diff
@@ -379,3 +379,61 @@ def cancel_booking(self, client, booking_id, status_code, json_error=None):
         if json_error:
             for key, msg in json_error.items():
                 assert response.json.get(key) == msg
+
+
+class ResetCollectiveBookingTest(PublicAPIRestrictedEnvEndpointHelper):
+    endpoint_url = "/v2/collective/adage_mock/bookings/{booking_id}/pending"
+    endpoint_method = "post"
+    default_path_params = {"booking_id": 1}
+
+    def test_reset_booking_not_linked_to_key(self, auth_client):
+        booking = factories.CollectiveBookingFactory()
+        with assert_attribute_does_not_change(booking, "status"):
+            self.reset_booking(auth_client, booking.id, status_code=404)
+
+    def test_can_reset_pending_booking(self, auth_client, pending_booking):
+        with assert_attribute_does_not_change(pending_booking, "status"):
+            # 1. get api key
+            # 2. get FF
+            # 3. get collective booking (no update triggered since status does not change)
+            booking_id = pending_booking.id
+            with assert_num_queries(3):
+                self.reset_booking(auth_client, booking_id, status_code=204)
```

Comment:
Point de détail, mais comme me le faisait remarquer @mageoffray sur une de mes PRs, il est préférable d'écrire les assert_num_queries ainsi (pour éviter que le prochain oublie de mettre à jour les commentaires)
```suggestion
            expected_num_queries = 1 # 1. get api key
            expected_num_queries += 1# 2. get FF
            expected_num_queries += 1 # 3. get collective booking (no update triggered since status does not change)
            booking_id = pending_booking.id
            with assert_num_queries(expected_num_queries):
                self.reset_booking(auth_client, booking_id, status_code=204)
```